lxd/storage/drivers/driver_common.go

```diff
@@ -0,0 +1,92 @@
+package drivers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/lxc/lxd/lxd/state"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/units"
+)
+
+type common struct {
+	name     string
+	config   map[string]string
+	getVolID func(volType VolumeType, volName string) (int64, error)
+	state    *state.State
+}
+
+func (d *common) init(state *state.State, name string, config map[string]string, volIDFunc func(volType VolumeType, volName string) (int64, error)) {
+	d.name = name
+	d.config = config
+	d.getVolID = volIDFunc
+	d.state = state
+}
+
+// validateVolumeCommonRules returns a map of volume config rules common to all drivers.
+func (d *common) validateVolumeCommonRules() map[string]func(string) error {
+	return map[string]func(string) error{
+		"security.shifted":    shared.IsBool,
+		"security.unmapped":   shared.IsBool,
+		"volatile.idmap.last": shared.IsAny,
+		"volatile.idmap.next": shared.IsAny,
```

Comment:
shifted/unmapped/volatile.idmap aren't going to be implemented at the driver level so should probably be kept away in the storage package

Reply:
@stgraber I have moved this into the utils file in the storage package, but have provided it as a function that the drivers can call, so that all of the actual validation checking is kept together in the driver. The reason for this is that as part of the translation of config between drivers, sometimes fields have to be removed. This logic is kept a lot simpler if the validation and removal of unknown fields is done in a single place. It also gives the drivers the opportunity to override/remove the validation of a common field if we need to in the future. Hopefully this change achieves both the simplicity of keeping it in one place as well as avoiding defining the common fields in the drivers package.